hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieFileGroupReader.java

```diff
@@ -256,6 +257,137 @@ public static RecordMergeMode getRecordMergeMode(Properties props) {
     return RecordMergeMode.valueOf(mergeMode);
   }
 
+  public static Builder builder() {
+    return new Builder<>();
+  }
+
+  public static class Builder<T> {
+
+    HoodieReaderContext<T> readerContext;
+    HoodieStorage storage;
+    String tablePath;
+    String latestCommitTime;
+    FileSlice fileSlice;
+    Schema dataSchema;
+    Schema requestedSchema;
+    Option<InternalSchema> internalSchemaOpt;
+    HoodieTableMetaClient hoodieTableMetaClient;
+    TypedProperties props;
+    HoodieTableConfig tableConfig;
+    long start;
+    long length;
+    boolean shouldUseRecordPosition = false;
+    long maxMemorySizeInBytes;
+    String spillableMapBasePath;
+    ExternalSpillableMap.DiskMapType diskMapType;
+    boolean isBitCaskDiskMapCompressionEnabled;
```

Comment:
wow. too many args.

Reply:
https://github.com/apache/hudi/pull/11449
https://github.com/apache/hudi/pull/11453
https://github.com/apache/hudi/pull/11455
followup prs to remove arguments